Anima/diffusion/mcm_estimator/animaMCMEstimatorImageFilter.hxx

```diff
@@ -207,48 +205,32 @@ MCMEstimatorImageFilter<InputPixelType, OutputPixelType>
     m_MCMCreators.resize(this->GetNumberOfThreads());
     for (unsigned int i = 0;i < this->GetNumberOfThreads();++i)
         m_MCMCreators[i] = this->GetNewMCMCreatorInstance();
-
-    // Use default known values for diffusivities (would be nice to store those values somewhere else)
-    m_AxialDiffusivityFixedValue = 1.71e-3;
-    m_RadialDiffusivity1FixedValue = 1.5e-4;
-    m_RadialDiffusivity2FixedValue = 1.5e-4;
     
-    std::cout << "Stick initial diffusivities:" << std::endl;
-    std::cout << " - Axial diffusivity: " << m_AxialDiffusivityFixedValue << " mm2/s," << std::endl;
-    std::cout << " - First radial diffusivity: " << m_RadialDiffusivity1FixedValue << " mm2/s," << std::endl;
-    std::cout << " - Second radial diffusivity: " << m_RadialDiffusivity2FixedValue << " mm2/s." << std::endl;
+    std::cout << "Initial diffusivities:" << std::endl;
+    std::cout << " - Axial diffusivity: " << m_AxialDiffusivityValue << " mm2/s," << std::endl;
+    std::cout << " - Radial diffusivity: " << m_RadialDiffusivityValue << " mm2/s," << std::endl;
+
+    if (m_ModelWithRestrictedWaterComponent)
+        std::cout << " - IRW diffusivity: " << m_IRWDiffusivityValue << " mm2/s," << std::endl;
+
+    if (m_ModelWithStaniszComponent)
+        std::cout << " - Stanisz diffusivity: " << m_StaniszDiffusivityValue << " mm2/s," << std::endl;
 
     // Setting up creators
     if (m_Optimizer == "levenberg")
         m_UseBoundedOptimization = true;
 
     for (unsigned int i = 0;i < this->GetNumberOfThreads();++i)
     {
-        m_MCMCreators[i]->SetAxialDiffusivityValue(m_AxialDiffusivityFixedValue);
+        m_MCMCreators[i]->SetAxialDiffusivityValue(m_AxialDiffusivityValue);
         m_MCMCreators[i]->SetFreeWaterDiffusivityValue(3.0e-3);
-        m_MCMCreators[i]->SetRadialDiffusivity1Value(m_RadialDiffusivity1FixedValue);
-        m_MCMCreators[i]->SetRadialDiffusivity2Value(m_RadialDiffusivity2FixedValue);
+        m_MCMCreators[i]->SetIRWDiffusivityValue(m_IRWDiffusivityValue);
+        m_MCMCreators[i]->SetStaniszDiffusivityValue(m_StaniszDiffusivityValue);
+        m_MCMCreators[i]->SetRadialDiffusivityValue(m_RadialDiffusivityValue);
```

Comment:
Why 3 radial diffusivities?

Reply:
You asked for it in a previous email (different value of radial diff than rd1+rd2 / 2). I can remove it if you wish it like that now. Initial radial diffs 1 and 2 are not used anymore in tensor (the only place where they were used) since we do grid search over the bounds of the parameters

Reply:
I do not understand your last point. The grid search is for initial parameters. But then I still want to estimate all 3 diffusivities in the tensor case.

Reply:
Ok you made me realize something here. I'm going to check the code again. In the end I might remove m_RadialDiffusivity to replace it back by rd1 + rd2 / 2